### PR TITLE
test: 3964 add HbarTransferMap _fromProtobuf unit coverage

### LINKED ISSUE #3964: [Beginner]: add unit tests for src/account/HbarTransferMap.js
### 🐥 Beginner Friendly

This issue is a great fit for contributors who are ready to explore the Hiero JavaScript
codebase a little more and take on slightly more independent work. We strongly recommend
completing a Good First Issue before tackling a Beginner Issue.

Beginner Issues often involve reading existing JavaScript code, understanding how
different parts of the SDK fit together, and making small, thoughtful updates
that follow established patterns.

### 👾 Description of the Issue

`src/account/HbarTransferMap.js` has **0% test coverage**. `HbarTransferMap` is a thin subclass of `ObjectMap` that maps `AccountId` keys to `Hbar` values. It is used in account balance and transaction record responses to represent HBAR transfers.

The class ([`src/account/HbarTransferMap.js`](../blob/main/src/account/HbarTransferMap.js)):

```js
class HbarTransferMap extends ObjectMap {
    constructor()  // passes AccountId.fromString as key factory to ObjectMap

    static _fromProtobuf(transfers: ITransferList): HbarTransferMap
    // Iterates transfers.accountAmounts (may be null → treated as [])
    // and calls _set(accountId, Hbar.fromTinybars(amount)) for each
}
```

The `_fromProtobuf` method has two paths:
1. `transfers.accountAmounts == null` → returns an empty map
2. `transfers.accountAmounts` is a non-empty array → populates the map with `AccountId` → `Hbar` entries

Both paths are currently untested.

### 💡 Proposed Solution

1. Create `test/unit/HbarTransferMap.js`
2. Write unit tests covering:
   - `_fromProtobuf` with `{ accountAmounts: null }` → returns empty map
   - `_fromProtobuf` with one or more `{ accountID, amount }` entries → correct `AccountId` → `Hbar` mappings
   - Verify that map entries can be retrieved by `AccountId` key (using `ObjectMap`'s `get` method)
3. All tests pass in both environments:
   - `task test:unit:node -- test/unit/HbarTransferMap.js`
   - `task test:unit:browser -- test/unit/HbarTransferMap.js`

### 👩‍💻 Implementation Steps

- Browse `test/unit/` for how similar data-class tests are structured
- Create `test/unit/HbarTransferMap.js`
- Import `HbarTransferMap` from `../../../src/account/HbarTransferMap.js`
- For proto input, construct a plain object: `{ accountAmounts: [{ accountID: { shardNum: 0, realmNum: 0, accountNum: 1 }, amount: 100 }] }`
- Use `Long.fromNumber(100)` if needed for the `amount` field (check how other tests handle `Long` values)
- Assert the returned map has an entry for the expected `AccountId` with the expected `Hbar` amount
- Test the null `accountAmounts` path separately
- Run both test environments and confirm all tests pass:
  - `task test:unit:node -- test/unit/HbarTransferMap.js`
  - `task test:unit:browser -- test/unit/HbarTransferMap.js`
- Keep changes localized to the new test file only — no production code changes

### ✅ Acceptance Criteria

- [ ] New test file added at `test/unit/HbarTransferMap.js`
- [ ] Null `accountAmounts` path covered
- [ ] Non-empty `accountAmounts` path covered
- [ ] Coverage for `src/account/HbarTransferMap.js` reaches 100%
- [ ] Both test environments pass with no failures:
  - `task test:unit:node -- test/unit/HbarTransferMap.js`
  - `task test:unit:browser -- test/unit/HbarTransferMap.js`
- [ ] `task lint` passes
- [ ] No changes to production code
- [ ] Review: All code review feedback addressed

### 📋 Step-by-Step Contribution Guide

- [ ] Comment `/assign` to request the issue and get assigned
- [ ] Fork the repository and create a branch from an updated main
- [ ] Make the requested changes and sign each commit using `-s -S`
- [ ] Push your branch and open a pull request

Read [Workflow Guide](manual/training/workflow.md) for step-by-step workflow guidance.  
Read [README.md](README.md) for setup instructions.

❗ Pull requests **cannot be merged** without signed commits (use `-s -S`).  
See the [Signing Guide](manual/training/signing.md).

### 🤔 Additional Information

If you have questions while working on this issue, feel free to ask!

You can reach the community and maintainers here:
[Hiero-SDK-JavaScript Discord](https://discord.com/channels/905194001349627914/1337424656138899537)

---

Part of Epic #3953 — improve unit test coverage across the SDK

### PR DESCRIPTION
**Pull request by labeebahmad201**
**Description**:
Add unit tests for `HbarTransferMap._fromProtobuf` to validate mapping behavior and improve file-level coverage for `src/account/HbarTransferMap.js`.
* Add test for `accountAmounts: null` returning an empty `HbarTransferMap`
* Add test for mapping a single protobuf transfer entry to `AccountId -> Hbar`
* Add test that `get()` works with both `AccountId` and string key forms
* Add test for multiple transfer entries, including negative tinybar values
* Add test covering constructor callback path (`_fromString`) to reach 100% file coverage for `HbarTransferMap.js`

**Related issue(s)**:
Fixes #3964

**Notes for reviewer**:
Coverage was verified for `src/account/HbarTransferMap.js` with all metrics at 100% (statements, branches, functions, lines).  

Because the repository test naming convention does not match Vitest defaults, coverage was checked by temporarily renaming the test file to `*.test.js` during the local run.

**Checklist**
- [x] Documented (Code comments)
- [x] Tested (unit)

### CODE REVIEW

**Comment by labeebahmad201** [comment]
Could a maintainer please add me as assignee? The assignee-check workflow is failing.

File: test/unit/account/HbarTransferMap.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: Apache-2.0
+
+import { describe, it, expect } from "vitest";
```
Comment (venilinvasilev) [inline test/unit/account/HbarTransferMap.js]: Hey, we do not need to import these here as they are auto imported in every test file. You can safely remove these.

Comment (labeebahmad201) [reply test/unit/account/HbarTransferMap.js]: @venilinvasilev this has been removed.

**Comment by venilinvasilev** [comment]
Hey @labeebahmad201. Thanks for your contribution. You will need to sign all of your commits in this PR so the DCO check passes.

**Comment by labeebahmad201** [comment]
> Hey @labeebahmad201. Thanks for your contribution. You will need to sign all of your commits in this PR so the DCO check passes.

@venilinvasilev Apparently link containing signing instructions mentioned in #3963 is returning 404. Please look into it.
Link -> `https://github.com/hiero-ledger/hiero-sdk-js/issues/manual/training/signing.md`

Screenshot 2026-04-30 at 10 20 39 PM

**Comment by venilinvasilev** [comment]
> > Hey @labeebahmad201. Thanks for your contribution. You will need to sign all of your commits in this PR so the DCO check passes.
> 
> @venilinvasilev Apparently link containing signing instructions mentioned in #3963 is returning 404. Please look into it. Link -> `https://github.com/hiero-ledger/hiero-sdk-js/issues/manual/training/signing.md`
> 
> Screenshot 2026-04-30 at 10 20 39 PM
The correct link is:
https://github.com/hiero-ledger/hiero-sdk-js/manual/issues/training/signing.md

**Comment by venilinvasilev** [comment]
Hey @labeebahmad201. I hope your doing well. I want to kindly remind you to re-sign your commits in this PR, so we can proceed with reviewing it withing the next 2 days, otherwise we will close it and unassign you from the issue.
You can refer to this [guide](https://github.com/hiero-ledger/hiero-sdk-js/blob/main/manual/training/signing.md) on how to sign your commits.
Thanks!

**Comment by venilinvasilev** [comment]
Closing this due to inactivity
